ui/js/models/resultset.js

```diff
@@ -227,12 +227,56 @@ treeherder.factory(
              },
 
              triggerNewJobs: function(repoName, resultset_id, buildernames, decisionTaskID) {
-                 var uri = resultset_id + '/trigger_runnable_jobs/';
-                 var data = {
-                     "requested_jobs": buildernames,
-                     "decision_task_id": decisionTaskID
-                 };
-                 return $http.post(thUrl.getProjectUrl("/resultset/", repoName) + uri, data);
-             }
+                 let tc = thTaskcluster.client();
+                 let queue = new tc.Queue();
+                 let url;
+                 try {
+                     url = queue.buildSignedUrl(queue.getLatestArtifact,
+                                                decisionTaskID,
+                                                'public/full-task-graph.json');
+                 } catch (e) {
+                     let errorMsg = e.message;
+                     if (errorMsg === 'credentials must be given') {
+                         errorMsg = 'Missing Taskcluster credentials! Please log out and back in again.';
+                     }
+                     throw new Error(errorMsg);
+                 }
+                 return $http.get(url).then(function(resp) {
+                     let graph = resp.data;
+                     let tclabels = _.intersection(buildernames, _.keys(graph));
+                     let bbnames = _.difference(buildernames, tclabels);
+                     return $q.all([
+                         $q.resolve().then(function() {
+                             if (bbnames.length === 0) {
+                                 return;
+                             }
+                             let uri = resultset_id + '/trigger_runnable_jobs/';
```

Comment:
I think it would be more clear to just tack this at the end of the call to $http.post, instead of creating a temporary (also `uri` is a misleading name).